src/java/org/apache/cassandra/dht/BootStrapper.java

```diff
@@ -72,6 +90,33 @@ public BootStrapper(InetAddressAndPort address,
         this.strictMovements = strictMovements;
     }
 
+    public void init()
+    {
+        addProgressListener((tag, event) -> {
```

Comment:
what do you think about changing init() to accept listeners... (might be on varargs)

in StorageService we do

        bootstrapper.init();
        bootstrapper.addProgressListener(progressSupport);

could not we do all of this in init() ?

Reply:
btw is it safe if somebody, by accident, calls this method twice? what would happen then?

Reply:
We can have only one bootstrap process running, there is a check for that:

```
        BootStrapper bootstrapper = new BootStrapper(getBroadcastAddressAndPort(), metadata, movements, strictMovements);
        boolean res = ongoingBootstrap.compareAndSet(null, bootstrapper);
        if (!res)
            throw new IllegalStateException("Bootstrap can be started exactly once, but seems to have already started: " + bootstrapper);
```

Anyway, I've moved the `init` to the constructor, so it should be clearer now.